File: admin-dashboard/src/api/supportContact.ts
import { api } from './client';

export interface SupportContactSettings {
  feedback_whatsapp_url: string;
  feedback_imessage_url: string;
  feedback_instagram_url: string;
  partnership_whatsapp_url: string;
  partnership_imessage_url: string;
  created_at: string;
  updated_at: string;
}

export interface SupportContactSettingsUpdatePayload {
  feedback_whatsapp_url: string;
  feedback_imessage_url: string;
  feedback_instagram_url: string;
  partnership_whatsapp_url: string;
  partnership_imessage_url: string;
}

export const getAdminSupportContactSettings = async () => {
  const response = await api.get('/support-contact/admin');
  return response.data as SupportContactSettings;
};

export const updateAdminSupportContactSettings = async (payload: SupportContactSettingsUpdatePayload) => {
  const response = await api.put('/support-contact/admin', payload);
  return response.data as SupportContactSettings;
};

File: backend/app/schemas/support_contact.py
"""Schemas for support and partnership contact settings."""
from datetime import datetime

from pydantic import BaseModel


class SupportContactSettingsResponse(BaseModel):
    feedback_whatsapp_url: str
    feedback_imessage_url: str
    feedback_instagram_url: str
    partnership_whatsapp_url: str
    partnership_imessage_url: str
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class SupportContactSettingsUpdateRequest(BaseModel):
    feedback_whatsapp_url: str
    feedback_imessage_url: str
    feedback_instagram_url: str
    partnership_whatsapp_url: str
    partnership_imessage_url: str
